naxsi_src/naxsi_runtime.c

```diff
@@ -3021,14 +3021,17 @@ ngx_http_naxsi_update_current_ctx_status(ngx_http_request_ctx_t*    ctx,
 #endif
 #endif
     {
-      ngx_str_t* ip = &r->connection->addr_text;
+      ngx_str_t ip;
+      ip.len  = r->connection->addr_text.len;
+      ip.data = ngx_pcalloc(r->pool, ip.len + 1);
```

Comment:
This pointer is never freed. call ngx_pfree

Reply:
Added ngx_pfree.